src/modm/platform/spi/sam/spi_master.hpp.in

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2021, Jeff McBride
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+#pragma once
+
+#include <modm/architecture/interface/spi_master.hpp>
+#include <modm/platform/gpio/connector.hpp>
+#include <modm/math/algorithm/prescaler.hpp>
+#include <modm/platform/clock/clockgen.hpp>
+
+namespace modm::platform {
+
+/**
+ * Serial peripheral interface on FLEXCOM{{ id }}.
+ *
+ * @author	Jeff McBride
+ * @ingroup	modm_platform_spi modm_platform_spi_{{id}}
+ */
+class SpiMaster{{ id }} : public modm::SpiMaster {
+	static uint8_t state;
+	static uint8_t count;
+	static void *context;
+	static ConfigurationHandler configuration;
+
+	// Work around a name collision between 'struct modm::Spi' and 'struct Spi'
+	// defined in the CMSIS headers
+	static modm_always_inline ::Spi*
+	Regs() { return (::Spi*) SPI{{ id }}; };
+
+	static modm_always_inline bool
+	isTransmitDataRegisterEmpty() {
+		return Regs()->SPI_SR & SPI_SR_TDRE;
+	}
+
+	static modm_always_inline bool
+	isReceiveDataRegisterFull() {
+		return Regs()->SPI_SR & SPI_SR_RDRF;
+	}
+
+	static modm_always_inline void
+	write(uint8_t data) {
+		Regs()->SPI_TDR = data;
+	}
+
+	static modm_always_inline uint8_t
```

Comment:
let's remove all the `modm_always_inline` in favor of just `inline`. Letting the compiler decide is better. Not sure why we have such an inlining fetish in modm…

Reply:
Well I suppose single instruction functions are always best inlined? But also I suspect the compiler will always inline them either way.